### ISSUE TITLE
error in the function _process_property_change of the class DatePicker

### ISSUE THREAD

**Issue by akasolace** [pyviz/panel#421]
The following code throws an error:

```python
datepicker = pn.widgets.DatePicker(name='Date Picker', value=datetime.datetime(2019, 1, 1))

def dateChanged(event):
    print(datepicker .value)

datepicker .param.watch(dateChanged, 'value')
```

when looking into it seems that `function _process_property_change` of the class `DatePicker` tries to return `datetime.strptime(msg['value'][4:], '%b %d %Y')` whereas the value is already a date and not a string.  Hence to fix the error, I made the following change (commented existing line and replace it below)

```python
    def _process_property_change(self, msg):
        msg = super(DatePicker, self)._process_property_change(msg)
        if 'value' in msg:
            # msg['value'] = datetime.strptime(msg['value'][4:], '%b %d %Y')
            msg['value'] = datetime.combine(msg['value'], datetime.min.time())
        return msg
```

**Comment by slarrain** [comment]
I have this same problem. It looks like this bus has not been addressed yet.
Are there any plans on doing it? I'm loving the library but not being able to use the date picker is kind of a deal breaker for me.
Thanks to the Developers for all their hard work.

**Comment by jbednar** [comment]
Sorry we haven't gotten to this yet; we've got a lot of things to try to achieve at the same time!  PRs would be gratefully accepted in the meantime.

**Comment by akasolace** [comment]
@jbednar @philippjfr if you want me to do a PR with my proposed solution... I am happy to do it.
Let me know

**Comment by philippjfr** [comment]
@akasolace That would be hugely appreciated.

**Comment by akasolace** [comment]
@akasolace ok, I'll do that shortly

**Comment by gwbischof** [comment]
What is the current status of this issue? It is working for me, and I noticed the PR was closed.

**Comment by philippjfr** [comment]
@gwbischof There is now a PR in param to fix the parameter types (https://github.com/pyviz/param/pull/348/files), once that's merged we can follow up with another PR in panel.

**Comment by Darrrian** [comment]
Is there any current update to the issue? I still have the problem, with panel v 0.9.3 and python 3.8

**Comment by philippjfr** [comment]
That's a separate regression caused by a change in bokeh 2.0, it was fixed in https://github.com/holoviz/panel/pull/1187 and will be released in 0.9.4  very soon.
